presto-hive/src/main/java/io/prestosql/plugin/hive/GenericHiveRecordCursorProvider.java

```diff
@@ -73,6 +73,10 @@ public GenericHiveRecordCursorProvider(HdfsEnvironment hdfsEnvironment, DataSize
             TypeManager typeManager,
             boolean s3SelectPushdownEnabled)
     {
+        if (s3SelectPushdownEnabled) {
```

Comment:
My understanding is that when `s3SelectPushdownEnabled` is enabled and serde is present in `CSV_SERDES`,  `S3SelectRecordCursor` should be picked over `GenericHiveRecordCursor`. Right now in the code, this choice depends on the implicit iteration order of `Set<HiveRecordCursorProvider> cursorProviders`. i.e. if the iteration ordered picked `GenericHiveRecordCursorProvider` first, then it would be the one to create the cursor. The purpose of adding this commit was to make the preference explicit. 

But I realized that in cases when `s3SelectPushdownEnabled` is true and the serde is not in `CSV_SERDES`, the current code falls back to using GenericHiveRecordCursor.  The commit I added changes that. That's why I've removed this commit, since the code works right now. But I think we shouldn't rely on the set's iteration order. What's your take here?